### PR TITLE
[HGCAL] MultiCluster to CaloParticle associator

### PR DESCRIPTION
**Pull request by lecriste**
#### PR description:

This PR introduces an associator that can be consumed by several analyzers, avoiding code duplication. This PR supersedes PR#32941, no changes are expected in the output.

#### PR validation:

`runTheMatrix -l limited`

### CODE REVIEW

**Comment by perrotta** [comment]
please test

File: SimDataFormats/Associations/plugins/MCToCPAssociatorEDProducer.cc [hunk 1]
```diff
@@ -0,0 +1,94 @@
+//
```
Comment (slava77) [inline SimDataFormats/Associations/plugins/MCToCPAssociatorEDProducer.cc:1]: this seems to grow  SimDataFormats/Associations/plugins.

@cms-sw/core-l2 is this a supported case or is there a chance for problems to have /plugins in a DataFormats?

Comment (makortel) [reply SimDataFormats/Associations/plugins/MCToCPAssociatorEDProducer.cc:1]: > is this a supported case or is there a chance for problems to have /plugins in a DataFormats?

It is not, plugins are not allowed in DataFormats (as mentioned in https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideCreatingNewProducts#Producing_the_EDProduct
> **NOTE: The producer must not be in the same package as the EDProduct.**

)

Comment (lecriste) [reply SimDataFormats/Associations/plugins/MCToCPAssociatorEDProducer.cc:1]: @makortel shall I move it here?
https://github.com/cms-sw/cmssw/tree/716967537b696c443b4a095f0008902ee692dc9f/SimCalorimetry/HGCalAssociatorProducers/plugins

Comment (makortel) [reply SimDataFormats/Associations/plugins/MCToCPAssociatorEDProducer.cc:1]: @lecriste By quick look that looks like a good place, thanks.

**Comment by srimanob** [comment]
Please test

File: SimDataFormats/Associations/plugins/MCToCPAssociatorEDProducer.cc (hunk 1 above)
Comment (lecriste) [reply SimDataFormats/Associations/plugins/MCToCPAssociatorEDProducer.cc]: OK, I moved the plugin introduced by this PR.
Will move the rest in another PR soon.

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/33384	9572be0 
- the reconstruction category part is almost trivial addition of a typedef in DataFormats/ParticleFlowReco/interface/HGCalMultiCluster.h

**Comment by civanch** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

**Comment by qliphy** [comment]
+1

**Pull request merged**